tests/Application/Actions/Donations/UpdateTest.php

```diff
@@ -1657,24 +1657,13 @@ public function testAddDataSuccessWithCashBalanceAutoconfirm(): void
         $container = $app->getContainer();
 
         $donation = $this->getTestDonation();
-        $donation->setDonorCountryCode('GB');
-        $donation->setCurrencyCode('GBP');
-        $donation->setGiftAid(true);
-        $donation->setTipAmount('0');
-        $donation->setTipGiftAid(false);
-        $donation->setDonorHomeAddressLine1('99 Updated St');
-        $donation->setDonorHomePostcode('X1 1XY');
-        $donation->setDonorFirstName('Saul');
-        $donation->setDonorLastName('Williams');
-        $donation->setDonorEmailAddress('saul@example.com');
-        $donation->setTbgComms(true);
-        $donation->setCharityComms(false);
-        $donation->setChampionComms(false);
-        $donation->setDonorBillingAddress('Y1 1YX');
+        $donation = $this->prepareDonationBasics($donation);
```

Comment:
Liking moving all these setter calls into a function, but:
- it seems to be overwriting many of the properties set in `\MatchBot\Tests\Application\DonationTestDataTrait::getTestDonation` making me thing we shouldn't need to use both.
- it needlessly returns the same object that was passed to it, which I think is a bit misleading. Better to either have it create an object (maybe by calling getTestDonation inside the function) or operate on a passed-in object and return void.

Reply:
Good spot – I obviously misdiagnosed the problems with tests early on and massively overestimated how much of this we needed. I guess some of the existing`UpdateTest` scenarios also pre-dated us having a helper that's as complete as the one we do now, so I started off copying work which already had redundant stuff.

Simplified it significantly & removed some redundant assertions on https://github.com/thebiggive/matchbot/pull/550/commits/9ac27612d06da705e672a915244e5022be5a6bce

Unfortunately psalm's baseline updates have gone into a tailspin with my local so none of the commands give me a way to update it, and I've run out of time for the week to try to placate it.

Reply:
Happy to have a look at it with you whenever you're next working. If there's ever an issue updating the baseline feel free to make the PR with the baseline not updated and ask me to update it. Another option is to ssh into a CircleCI executor and run the update there. We could also change the Psalm config to not require the baseline to be updated all the time, and just update it when we feel like it - although it is nice to see issues disappearing from the basline as we work on stuff.